l10n_th_withholding_tax_cert/models/withholding_tax_cert.py

```diff
@@ -105,14 +130,45 @@ class WithholdingTaxCert(models.Model):
         copy=False,
     )
 
+    def _get_domain_payment_id(self):
+        domain = (
+            "[('partner_id', '=', supplier_partner_id), ('wt_cert_ids', '=', False)]"
+        )
+        return domain
+
+    def _get_domain_move_id(self):
+        domain = [
+            ("journal_id.type", "=", "general"),
+            ("wt_cert_ids", "=", False),
+            ("state", "=", "posted"),
+        ]
+        return domain
+
+    @api.model
+    def _default_company_id(self):
+        return self.env.company
+
     @api.onchange("payment_id")
-    def _onchange_payment_id(self):
+    def _onchange_payment_move_id(self):
         """ Prepare withholding cert """
         wt_account_ids = self._context.get("wt_account_ids", [])
-        self.date = self.payment_id.payment_date
-        self.supplier_partner_id = self.payment_id.partner_id
         # Hook to find wt move lines
-        wt_move_lines = self._get_wt_move_line(self.payment_id, wt_account_ids)
+        wt_move_lines = self._get_wt_move_line(
+            self.payment_id, self.move_id, wt_account_ids
+        )
+        partner_id = self.payment_id.partner_id or self.move_id.partner_id
+        # Create Direct Journal Entry and Same Partner in line.
+        if self.move_id and self.move_id.type == "entry":
+            partner = wt_move_lines.mapped("partner_id")
+            if len(partner) == 1:
```

Comment:
If len(partner) > 1: throw exception "More than one partners in journal items."

Reply:
Solved